app/models/concerns/specimen_role.rb

```diff
@@ -18,9 +23,14 @@ def specimen_role_descriptions
       YAML.load_file(File.join("app", "models", "config", "specimen_roles.yml"))["specimen_roles"]
     end
 
+    def specimen_role_description(id)
+      specimen_role_id = id || ""
+      "#{specimen_role_id.upcase} - #{specimen_role_descriptions[specimen_role_id] || ""}"
+    end
```

Comment:
I understand that the `specimen_role` attribute is optional? Then maybe we should return nil or an empty string in that case? For example:

```suggestion
    def specimen_role_description(id)
      "#{id.upcase} - #{specimen_role_descriptions[id]}" if id
    end
```